oci/spec_opts_windows.go

```diff
@@ -100,3 +101,52 @@ func escapeAndCombineArgs(args []string) string {
 	}
 	return strings.Join(escaped, " ")
 }
+
+func WithDevices(devicePath, containerPath, permissions string) SpecOpts {
```

Comment:
I'm a little doubtful about whether `oci.WithDevices` is the right API here, or if it should be `oci.WithWindowsDevices` and `oci.WithLinuxDevices`, and the splitting/guessing logic could be moved into `pkg/cri/opts.WithDevices`.

I guess it depends on the use-case for `oci.WithDevices`... Would direct callers know what kind of devices they want at the OCI level, or would they expect to provide the same structure as CRI and docker, leaving the OCI meaning up to the recipient?

Reply:
The more I think about it, the more I want to make this change. The main thing holding me back is that the other platform code uses `WithDevices` with this signature, and I kind-of like the opportunity to close a platform-specific API difference.

Reply:
After sleeping on it, perhaps the API should be `oci.WithWindowsDevices(IDType, ID)` and `oci.WithLCOWDevices(DevicePath)`? Unlike the `oci.WithDevices` defined on the Unix platforms, we never use 'containerPath' or 'permissions', so why take them as parameters?

That said, I'm unsure if that's burning in a behaviour aspect at the wrong level, i.e. on Windows builds, is the `oci` library *always* creating WCOW or LCOW setups, and *never* actual-Linux setups?

Reply:
> on Windows builds, is the oci library always creating WCOW or LCOW setups, and never actual-Linux setups?

On reflection, this is probably intrinsically true in containerd, but not in the client code, as it may be a Windows-based client talking to a Linux-based containerd. I'm pretty sure (modulo breakfast) that this code is running in containerd, since at this layer we're making decisions based on `osinterface.OS` in CRI, and oci/utils_unix.go implements `getDevices` which also pokes at the local system assuming it's the host. So I can continue with that assumption.

Reply:
`oci.WithWindowsDevices(IDType, ID)` - I'm in this boat personally. I don't see why we would have a generic API that makes virtually no sense for 2 of the params for Windows and the 1'st param actually has to be parsed and isnt a path at all... However, there are more paths than just class id's. But something like IDType=PNP and ID= a pnp device id, or IDType=A VF, ID=..., seems to also fit the mold just fine. Which means that IDType really solves your other question. If it really came down to it you could do IDType=HostPNP or GuestPNP or whatever.

Reply:
Resolving as I've rewritten this per https://github.com/containerd/containerd/pull/6618#issuecomment-1060203801